Azure.Core.Pole.Tests/TypeSpecGenerator.cs

```diff
@@ -2,20 +2,65 @@
 // Licensed under the MIT License.
 
 using Azure.Core.Pole.Tooling;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
 using NUnit.Framework;
+using System;
 using System.IO;
+using System.Linq;
+using System.Reflection;
 
 namespace Azure.Core.Pole.Tests
 {
     public partial class TypeSpecGeneratorTests
     {
         [Test]
-        public void Basics()
+        public void DogModelGeneration()
         {
-            var source = Path.Combine(".", "tsp", "simple.tsp");
-            var destination = Path.Combine(Path.GetTempPath());
-            TypeSpecGenerator.Generate(source, destination);
-            Directory.Delete(destination, true);
+            // Read the TypeSpec content directly
+            string typeSpecContent = @"model Dog {
```

Comment:
read tsp for the dog.tsp file (as it was before). What I meant is that C# files should not be generated by writing them to disk. THey should be generated in-memory only